packages/shorebird_cli/lib/src/commands/patch/patch_ios_command.dart

```diff
@@ -0,0 +1,332 @@
+    );
+    final String shorebirdFlutterRevision;
+    try {
+      shorebirdFlutterRevision = await getShorebirdFlutterRevision();
+      flutterRevisionProgress.complete();
+    } catch (error) {
+      flutterRevisionProgress.fail('$error');
+      return ExitCode.software.code;
+    }
+
+    if (release.flutterRevision != shorebirdFlutterRevision) {
+      logger
+        ..err('''
+Flutter revision mismatch.
+
+The release you are trying to patch was built with a different version of Flutter.
+
+Release Flutter Revision: ${release.flutterRevision}
+Current Flutter Revision: $shorebirdFlutterRevision
+''')
+        ..info(
+          '''
+Either create a new release using:
+  ${lightCyan.wrap('shorebird release')}
+
+Or downgrade your Flutter version and try again using:
+  ${lightCyan.wrap('cd ${ShorebirdEnvironment.flutterDirectory.path}')}
+  ${lightCyan.wrap('git checkout ${release.flutterRevision}')}
+
+Shorebird plans to support this automatically, let us know if it's important to you:
+https://github.com/shorebirdtech/shorebird/issues/472
+''',
+        );
+      return ExitCode.software.code;
+    }
+
+    if (dryRun) {
+      logger
+        ..info('No issues detected.')
+        ..info('The server may enforce additional checks.');
+      return ExitCode.success.code;
+    }
+
+    final size = formatBytes(aotFile.statSync().size);
+
+    final summary = [
+      '''📱 App: ${lightCyan.wrap(app.displayName)} ${lightCyan.wrap('(${app.id})')}''',
+      if (flavor != null) '🍧 Flavor: ${lightCyan.wrap(flavor)}',
+      '📦 Release Version: ${lightCyan.wrap(releaseVersion)}',
+      '📺 Channel: ${lightCyan.wrap(channelName)}',
+      '''🕹️  Platform: ${lightCyan.wrap(platform)} ${lightCyan.wrap('[arm64 ($size)]')}''',
+    ];
+
+    logger.info(
+      '''
+
+${styleBold.wrap(lightGreen.wrap('🚀 Ready to publish a new patch!'))}
+
+${summary.join('\n')}
+''',
+    );
+
+    final needsConfirmation = !force;
+    if (needsConfirmation) {
+      final confirm = logger.confirm('Would you like to continue?');
+
+      if (!confirm) {
+        logger.info('Aborting.');
+        return ExitCode.success.code;
+      }
+    }
+
+    final Patch patch;
+    try {
+      patch = await createPatch(releaseId: release.id);
+    } catch (e) {
+      return ExitCode.software.code;
+    }
+
+    final codePushClient = buildCodePushClient(
+      httpClient: auth.client,
+      hostedUri: hostedUri,
+    );
+
+    // TODO(bryanoltman): check for asset changes
+
+    final createArtifactProgress = logger.progress('Uploading artifacts');
+    try {
+      await codePushClient.createPatchArtifact(
+        patchId: patch.id,
+        artifactPath: aotFile.path,
+        arch: 'arm64',
+        platform: 'ios',
+        hash: _hashFn(await aotFile.readAsBytes()),
+      );
+    } catch (error) {
+      createArtifactProgress.fail('$error');
+      return ExitCode.software.code;
+    }
+    createArtifactProgress.complete();
+
+    Channel? channel;
```

Comment:
Does the channel code do anything?

Reply:
yeah it's needed to promote the patch to the stable channel since the act of creating a patch doesn't immediately make it available atm. We can change that and remove the concept of a channel entirely but it would involve backend changes as well.

Reply:
Removed the argument. We now always use "stable".